projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -12,12 +12,89 @@ export function cloneArray(array, deep?: boolean) {
     }
     return arr;
 }
+
 /**
+ * Deep clones all first level keys of Obj2 and merges them to Obj1
+ * @param obj1 Object to merge into
+ * @param obj2 Object to merge from
+ * @returns Obj1 with merged cloned keys from Obj2
+ * @hidden
+ */
+export function mergeObjects(obj1: {}, obj2: {}) {
+    if (obj1 === null || obj1 === undefined) {
+        return cloneObject(obj2);
+    }
+
+    if (obj2 === null || obj2 === undefined) {
+        return obj1;
+    }
+
+    if (!isObject(obj1)) {
+        throw new Error(`Should provide objects to mergeObjects method. ${obj1} is not an object!`);
+    }
+
+    if (!isObject(obj2)) {
+        throw new Error(`Should provide objects to mergeObjects method. ${obj2} is not an object!`);
```

Comment:
And this can probably just 
```suggestion
        return obj1;
```
and skip the 2nd check